### ISSUE TITLE
Configure fails

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#8501]
Migrated from [rt.perl.org#39636](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=39636) (status was 'resolved')

Searchable as RT39636$

**Comment by p5pRT** [comment]
### From esserm@us.ibm.com

Hello\,

I have a problem with a 64\-bit installation of Perl 5\.8\.8 on Solaris 10\. 

sh Configure \-Duse64bitall \-Duseithreads \-Dusethreads \-Dusemultiplicity 
\-Dcc='gcc \-m64 \-mcpu=v9'

does its thing\, then \.\.\. 

Use which C compiler? \[gcc \-m64 \-mcpu=v9\]
/usr/ccs/bin/as&#8203;: error&#8203;: unknown option 'xarch=generic64'
usage&#8203;: /usr/ccs/bin/as \[\-V\] \[\-Q\{y\,n\}\] \[\-q\] \[\-s\]
&nbsp;          \[\-S\] \[\-K \{pic\,PIC\}\] \[\-o objfile\] \[\-L\] \[\-T\]
&nbsp;          \[\-P \[\[\-Yc\,path\] \[\-Ipath\] \[\-Dname\] \[\-Dname=def\] \[\-Uname\]\]\.\.\.\]
&nbsp;          \[\-m \[\-Ym\,path\]\] \[\-n\] \[\-ul\] \[\-xF\] 
\[\-xarch=\{v7\,v8\,v8a\,v8plus\,v8plusa\,v8plusb\,v9\,v9a\,v9b\}\] 
\[\-xcode=\{pic13\,pic32\}\] file\.s\.\.\.
Uh\-oh\, the C compiler 'gcc \-m64 \-mcpu=v9' doesn't seem to be working\.

\[message repeated twice\, then Configure aborts\.\] 

Apparently\, 'as' does not support the xarch=generic64 option\, despite its 
man page mentioning it\. 

Any help appreciated\! Google yielded nothing\. Thanks\!\!



---
Configuration details&#8203;:

bash\-3\.00\# pwd
/export/home/esserma/perl\-5\.8\.8
bash\-3\.00\# \./myconfig
bash&#8203;: \./myconfig&#8203;: No such file or directory           \[this is a 
contradiction to the INSTALL document\, by the way\]
bash\-3\.00\# uname \-a
SunOS \[server name left out\] 5\.10 Generic\_118833\-03 sun4u sparc 
SUNW\,Sun\-Fire\-880
bash\-3\.00\# gcc \-v
Reading specs from 
/usr/local/lib/gcc\-lib/sparc\-sun\-solaris2\.10/3\.3\.2/specs
Configured with&#8203;: \.\./configure \-\-with\-as=/usr/ccs/bin/as 
\-\-with\-ld=/usr/ccs/bin/ld \-\-disable\-nls
Thread model&#8203;: posix
gcc version 3\.3\.2
bash\-3\.00\# which as
/usr/ccs/bin/as
bash\-3\.00\# as \-V
as&#8203;: Sun Compiler Common 10 Patch 05/06/2005
bash\-3\.00\# getconf \-a | grep xarch
POSIX\_V6\_LP64\_OFF64\_CFLAGS&#8203;:     \-xarch=generic64
POSIX\_V6\_LP64\_OFF64\_LDFLAGS&#8203;:    \-xarch=generic64
POSIX\_V6\_LPBIG\_OFFBIG\_CFLAGS&#8203;:   \-xarch=generic64
POSIX\_V6\_LPBIG\_OFFBIG\_LDFLAGS&#8203;:  \-xarch=generic64
XBS5\_LP64\_OFF64\_CFLAGS&#8203;:         \-xarch=generic64
XBS5\_LP64\_OFF64\_LDFLAGS&#8203;:        \-xarch=generic64
XBS5\_LP64\_OFF64\_LINTFLAGS&#8203;:      \-xarch=generic64
XBS5\_LPBIG\_OFFBIG\_CFLAGS&#8203;:       \-xarch=generic64
XBS5\_LPBIG\_OFFBIG\_LDFLAGS&#8203;:      \-xarch=generic64
XBS5\_LPBIG\_OFFBIG\_LINTFLAGS&#8203;:    \-xarch=generic64
\_XBS5\_LP64\_OFF64\_CFLAGS&#8203;:        \-xarch=generic64
\_XBS5\_LP64\_OFF64\_LDFLAGS&#8203;:       \-xarch=generic64
\_XBS5\_LP64\_OFF64\_LINTFLAGS&#8203;:     \-xarch=generic64
\_XBS5\_LPBIG\_OFFBIG\_CFLAGS&#8203;:      \-xarch=generic64
\_XBS5\_LPBIG\_OFFBIG\_LDFLAGS&#8203;:     \-xarch=generic64
\_XBS5\_LPBIG\_OFFBIG\_LINTFLAGS&#8203;:   \-xarch=generic64


Any advice is welcome\. 

Thanks 

Marc Esser
IBM Global Services
esserm@&#8203;us\.ibm\.com

**Comment by p5pRT** [comment]
@rgs - Status changed from 'new' to 'rejected'

**Comment by p5pRT** [comment]
### From @rgs

Marc Esser \(via RT\) wrote&#8203;:
> I have a problem with a 64\-bit installation of Perl 5\.8\.8 on Solaris 10\. 
> 
> sh Configure \-Duse64bitall \-Duseithreads \-Dusethreads \-Dusemultiplicity 
> \-Dcc='gcc \-m64 \-mcpu=v9'

You shouldn't include flags in the cc variable\, it's only for the
name/path to the compiler\. Use ccflags for that\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @doughera88

On Mon\, 26 Jun 2006\, Marc Esser wrote&#8203;:

> \# New Ticket Created by  Marc Esser 
> \# Please include the string&#8203;:  \[perl \#39636\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=39636 >
> 
> 
> Hello\,
> 
> I have a problem with a 64\-bit installation of Perl 5\.8\.8 on Solaris 10\. 
> 
> sh Configure \-Duse64bitall \-Duseithreads \-Dusethreads \-Dusemultiplicity 
> \-Dcc='gcc \-m64 \-mcpu=v9'
> 
> does its thing\, then \.\.\. 
> 
> Use which C compiler? \[gcc \-m64 \-mcpu=v9\]
> /usr/ccs/bin/as&#8203;: error&#8203;: unknown option 'xarch=generic64'
> usage&#8203;: /usr/ccs/bin/as \[\-V\] \[\-Q\{y\,n\}\] \[\-q\] \[\-s\]
>           \[\-S\] \[\-K \{pic\,PIC\}\] \[\-o objfile\] \[\-L\] \[\-T\]
>           \[\-P \[\[\-Yc\,path\] \[\-Ipath\] \[\-Dname\] \[\-Dname=def\] \[\-Uname\]\]\.\.\.\]
>           \[\-m \[\-Ym\,path\]\] \[\-n\] \[\-ul\] \[\-xF\] 
> \[\-xarch=\{v7\,v8\,v8a\,v8plus\,v8plusa\,v8plusb\,v9\,v9a\,v9b\}\] 
> \[\-xcode=\{pic13\,pic32\}\] file\.s\.\.\.
> Uh\-oh\, the C compiler 'gcc \-m64 \-mcpu=v9' doesn't seem to be working\.
> 
> \[message repeated twice\, then Configure aborts\.\] 
> 
> Apparently\, 'as' does not support the xarch=generic64 option\, despite its 
> man page mentioning it\. 

This looks to be a problem in your gcc configuration\.  According to the 
Solaris 10 man page \(from http&#8203;://docs\.sun\.com\) the \-xarch=generic64 option 
to the assembler /usr/ccs/bin/as is only valid for the Solaris/x86 
platform\, not the SPARC platform\.

> bash\-3\.00\# \./myconfig
> bash&#8203;: \./myconfig&#8203;: No such file or directory           \[this is a 
> contradiction to the INSTALL document\, by the way\]

No\, it isn't\.  Here's the relevant quote from the INTSALL file&#8203;:

&nbsp;    If the build fails too early to run perlbug uninstalled\, then please
&nbsp;    B\<run> the C\<\./myconfig> shell script\, and mail its output along with
&nbsp;    an accurate description of your problem to perlbug@&#8203;perl\.org

&nbsp;    If Configure itself fails\, and does not generate a config\.sh file
&nbsp;    \(needed to run C\<\./myconfig>\)\, then please mail perlbug@&#8203;perl\.org the
&nbsp;    description of how Configure fails along with details of your system
&nbsp;    \- for example the output from running C\<uname \-a>


> bash\-3\.00\# getconf \-a | grep xarch
> POSIX\_V6\_LP64\_OFF64\_CFLAGS&#8203;:     \-xarch=generic64

\[etc\]

These are flags for cc\.  I think Sun's C compiler indeed does accept the 
\-xarch=generic64 flag and translate it\, as appropriate\.  For the SPARC 
architecture\, it translates it to \-xarch=v9\, which is\, in fact\, a valid 
/usr/ccs/bin/as flag\.

In short\, I'd go back and check over your gcc configuration\.  It doesn't 
appear to be able to compile 64\-bit applications\.  If there's nothing that 
can be done on the gcc configuration end\, we can probably work around it 
somehow in Configure or hints/solaris\_2\.sh\, but my first recommendation 
would be to try fixing gcc\.

\-\- 
&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From @doughera88

On Tue\, 27 Jun 2006\, Rafael Garcia\-Suarez wrote&#8203;:

> Marc Esser \(via RT\) wrote&#8203;:
> > I have a problem with a 64\-bit installation of Perl 5\.8\.8 on Solaris 10\. 
> > 
> > sh Configure \-Duse64bitall \-Duseithreads \-Dusethreads \-Dusemultiplicity 
> > \-Dcc='gcc \-m64 \-mcpu=v9'
> 
> You shouldn't include flags in the cc variable\, it's only for the
> name/path to the compiler\. Use ccflags for that\.

While generally good advice\, including extra flags in cc ought to work 
too\.  I know I've used it on many occasions\.  \(Although it shouldn't be 
necessary here \-\- with \-Duse64bitall\, the hints/solaris\_2\.sh file should 
be able to figure out the appropriate flags\.\)

\-\- 
&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From esserm@us.ibm.com

Andy\,

thanks for your tips\. I've done some more digging\, it looks like the 
bottom of the problem is that 
getconf \-a | grep xarch 
on my system erroneously reports xarch=generic64 even though this is a 
Sparc machine\. When doing the same on a similar system\, but with Solaris 
9\, I get the correct answer of xarch=v9\. 

Our system builders will look into that\. In the meantime I have tried 
building gcc 4\.1\.1\, but can't test/install it due to a problem with 
dejagnu\. 

That said\, I would really appreciate if you could help me with a 
hints\-based workaround\. 

Thanks\!

Marc Esser
IBM Global Services
esserm at us dot ibm dot com




"Andy Dougherty via RT" \<perlbug\-followup@&#8203;perl\.org> 
06/27/2006 07&#8203;:32 AM
Please respond to
perlbug\-followup@&#8203;perl\.org


To
Marc Esser/Encinitas/IBM@&#8203;IBMUS
cc

Subject
Re&#8203;: \[perl \#39636\] Configure fails






On Mon\, 26 Jun 2006\, Marc Esser wrote&#8203;:

> \# New Ticket Created by  Marc Esser 
> \# Please include the string&#8203;:  \[perl \#39636\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=39636 >
> 
> 
> Hello\,
> 
> I have a problem with a 64\-bit installation of Perl 5\.8\.8 on Solaris 10\. 

> 
> sh Configure \-Duse64bitall \-Duseithreads \-Dusethreads \-Dusemultiplicity 
> \-Dcc='gcc \-m64 \-mcpu=v9'
> 
> does its thing\, then \.\.\. 
> 
> Use which C compiler? \[gcc \-m64 \-mcpu=v9\]
> /usr/ccs/bin/as&#8203;: error&#8203;: unknown option 'xarch=generic64'
> usage&#8203;: /usr/ccs/bin/as \[\-V\] \[\-Q\{y\,n\}\] \[\-q\] \[\-s\]
>           \[\-S\] \[\-K \{pic\,PIC\}\] \[\-o objfile\] \[\-L\] \[\-T\]
>           \[\-P \[\[\-Yc\,path\] \[\-Ipath\] \[\-Dname\] \[\-Dname=def\] \[\-Uname\]\]\.\.\.\]
>           \[\-m \[\-Ym\,path\]\] \[\-n\] \[\-ul\] \[\-xF\] 
> \[\-xarch=\{v7\,v8\,v8a\,v8plus\,v8plusa\,v8plusb\,v9\,v9a\,v9b\}\] 
> \[\-xcode=\{pic13\,pic32\}\] file\.s\.\.\.
> Uh\-oh\, the C compiler 'gcc \-m64 \-mcpu=v9' doesn't seem to be working\.
> 
> \[message repeated twice\, then Configure aborts\.\] 
> 
> Apparently\, 'as' does not support the xarch=generic64 option\, despite 
its 
> man page mentioning it\. 

This looks to be a problem in your gcc configuration\.  According to the 
Solaris 10 man page \(from http&#8203;://docs\.sun\.com\) the \-xarch=generic64 option 

to the assembler /usr/ccs/bin/as is only valid for the Solaris/x86 
platform\, not the SPARC platform\.

> bash\-3\.00\# \./myconfig
> bash&#8203;: \./myconfig&#8203;: No such file or directory           \[this is a 
> contradiction to the INSTALL document\, by the way\]

No\, it isn't\.  Here's the relevant quote from the INTSALL file&#8203;:

&nbsp;    If the build fails too early to run perlbug uninstalled\, then please
&nbsp;    B\<run> the C\<\./myconfig> shell script\, and mail its output along with
&nbsp;    an accurate description of your problem to perlbug@&#8203;perl\.org

&nbsp;    If Configure itself fails\, and does not generate a config\.sh file
&nbsp;    \(needed to run C\<\./myconfig>\)\, then please mail perlbug@&#8203;perl\.org the
&nbsp;    description of how Configure fails along with details of your system
&nbsp;    \- for example the output from running C\<uname \-a>


> bash\-3\.00\# getconf \-a | grep xarch
> POSIX\_V6\_LP64\_OFF64\_CFLAGS&#8203;:     \-xarch=generic64

\[etc\]

These are flags for cc\.  I think Sun's C compiler indeed does accept the 
\-xarch=generic64 flag and translate it\, as appropriate\.  For the SPARC 
architecture\, it translates it to \-xarch=v9\, which is\, in fact\, a valid 
/usr/ccs/bin/as flag\.

In short\, I'd go back and check over your gcc configuration\.  It doesn't 
appear to be able to compile 64\-bit applications\.  If there's nothing that 

can be done on the gcc configuration end\, we can probably work around it 
somehow in Configure or hints/solaris\_2\.sh\, but my first recommendation 
would be to try fixing gcc\.

\-\- 
&nbsp;    Andy Dougherty                               doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From esserm@us.ibm.com

ok\, I have changed my invocation to 

sh Configure \-Duse64bitall \-Duseithreads \-Dusethreads \-Dusemultiplicity 
\-Dcc=gcc \-Accflags='\-m64 \-mcpu=v9'

unfortunately\, I get the exact same result :\-/

Marc Esser
IBM Global Services
esserm@&#8203;us\.ibm\.com
Office \(858\) 795\-5593
Cell \(619\) 322\-6651




"Rafael Garcia\-Suarez via RT" \<perlbug\-followup@&#8203;perl\.org> 
06/27/2006 01&#8203;:23 AM
Please respond to
perlbug\-followup@&#8203;perl\.org


To
Marc Esser/Encinitas/IBM@&#8203;IBMUS
cc

Subject
Re&#8203;: \[perl \#39636\] Configure fails






Marc Esser \(via RT\) wrote&#8203;:
> I have a problem with a 64\-bit installation of Perl 5\.8\.8 on Solaris 10\. 

> 
> sh Configure \-Duse64bitall \-Duseithreads \-Dusethreads \-Dusemultiplicity 
> \-Dcc='gcc \-m64 \-mcpu=v9'

You shouldn't include flags in the cc variable\, it's only for the
name/path to the compiler\. Use ccflags for that\.

**Comment by p5pRT** [comment]
### From esserm@us.ibm.com

Andy\,

don't mean to clobber the list\, but I just found a workaround for this 
problem\. This got me past the message in configure \- so I should be good 
to go for now\. Maybe include that in the README file as a known issue??

http&#8203;://bugs\.opensolaris\.org/bugdatabase/view\_bug\.do?bug\_id=6315357


Marc Esser
IBM Global Services
esserm at us dot ibm dot com




Marc Esser/Encinitas/IBM@&#8203;IBMUS
Sent by&#8203;: Marc Esser
06/27/2006 03&#8203;:54 PM

To
perlbug\-followup@&#8203;perl\.org
cc

Subject
Re&#8203;: \[perl \#39636\] Configure fails





Andy\,

thanks for your tips\. I've done some more digging\, it looks like the 
bottom of the problem is that 
getconf \-a | grep xarch 
on my system erroneously reports xarch=generic64 even though this is a 
Sparc machine\. When doing the same on a similar system\, but with Solaris 
9\, I get the correct answer of xarch=v9\. 

Our system builders will look into that\. In the meantime I have tried 
building gcc 4\.1\.1\, but can't test/install it due to a problem with 
dejagnu\. 

That said\, I would really appreciate if you could help me with a 
hints\-based workaround\. 

Thanks\!

Marc Esser
IBM Global Services
esserm at us dot ibm dot com




"Andy Dougherty via RT" \<perlbug\-followup@&#8203;perl\.org> 
06/27/2006 07&#8203;:32 AM
Please respond to
perlbug\-followup@&#8203;perl\.org


To
Marc Esser/Encinitas/IBM@&#8203;IBMUS
cc

Subject
Re&#8203;: \[perl \#39636\] Configure fails






On Mon\, 26 Jun 2006\, Marc Esser wrote&#8203;:

> \# New Ticket Created by  Marc Esser 
> \# Please include the string&#8203;:  \[perl \#39636\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=39636 >
> 
> 
> Hello\,
> 
> I have a problem with a 64\-bit installation of Perl 5\.8\.8 on Solaris 10\. 

> 
> sh Configure \-Duse64bitall \-Duseithreads \-Dusethreads \-Dusemultiplicity 
> \-Dcc='gcc \-m64 \-mcpu=v9'
> 
> does its thing\, then \.\.\. 
> 
> Use which C compiler? \[gcc \-m64 \-mcpu=v9\]
> /usr/ccs/bin/as&#8203;: error&#8203;: unknown option 'xarch=generic64'
> usage&#8203;: /usr/ccs/bin/as \[\-V\] \[\-Q\{y\,n\}\] \[\-q\] \[\-s\]
>           \[\-S\] \[\-K \{pic\,PIC\}\] \[\-o objfile\] \[\-L\] \[\-T\]
>           \[\-P \[\[\-Yc\,path\] \[\-Ipath\] \[\-Dname\] \[\-Dname=def\] \[\-Uname\]\]\.\.\.\]
>           \[\-m \[\-Ym\,path\]\] \[\-n\] \[\-ul\] \[\-xF\] 
> \[\-xarch=\{v7\,v8\,v8a\,v8plus\,v8plusa\,v8plusb\,v9\,v9a\,v9b\}\] 
> \[\-xcode=\{pic13\,pic32\}\] file\.s\.\.\.
> Uh\-oh\, the C compiler 'gcc \-m64 \-mcpu=v9' doesn't seem to be working\.
> 
> \[message repeated twice\, then Configure aborts\.\] 
> 
> Apparently\, 'as' does not support the xarch=generic64 option\, despite 
its 
> man page mentioning it\. 

This looks to be a problem in your gcc configuration\.  According to the 
Solaris 10 man page \(from http&#8203;://docs\.sun\.com\) the \-xarch=generic64 option 

to the assembler /usr/ccs/bin/as is only valid for the Solaris/x86 
platform\, not the SPARC platform\.

> bash\-3\.00\# \./myconfig
> bash&#8203;: \./myconfig&#8203;: No such file or directory           \[this is a 
> contradiction to the INSTALL document\, by the way\]

No\, it isn't\.  Here's the relevant quote from the INTSALL file&#8203;:

&nbsp;    If the build fails too early to run perlbug uninstalled\, then please
&nbsp;    B\<run> the C\<\./myconfig> shell script\, and mail its output along with
&nbsp;    an accurate description of your problem to perlbug@&#8203;perl\.org

&nbsp;    If Configure itself fails\, and does not generate a config\.sh file
&nbsp;    \(needed to run C\<\./myconfig>\)\, then please mail perlbug@&#8203;perl\.org the
&nbsp;    description of how Configure fails along with details of your system
&nbsp;    \- for example the output from running C\<uname \-a>


> bash\-3\.00\# getconf \-a | grep xarch
> POSIX\_V6\_LP64\_OFF64\_CFLAGS&#8203;:     \-xarch=generic64

\[etc\]

These are flags for cc\.  I think Sun's C compiler indeed does accept the 
\-xarch=generic64 flag and translate it\, as appropriate\.  For the SPARC 
architecture\, it translates it to \-xarch=v9\, which is\, in fact\, a valid 
/usr/ccs/bin/as flag\.

In short\, I'd go back and check over your gcc configuration\.  It doesn't 
appear to be able to compile 64\-bit applications\.  If there's nothing that 

can be done on the gcc configuration end\, we can probably work around it 
somehow in Configure or hints/solaris\_2\.sh\, but my first recommendation 
would be to try fixing gcc\.

\-\- 
&nbsp;    Andy Dougherty                               doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From @smpeters

> \[esserm@&#8203;us\.ibm\.com \- Wed Jun 28 03&#8203;:54&#8203;:19 2006\]&#8203;:
> 
> Andy\,
> 
> don't mean to clobber the list\, but I just found a workaround for this 
> problem\. This got me past the message in configure \- so I should be good 
> to go for now\. Maybe include that in the README file as a known issue??
> 
> http&#8203;://bugs\.opensolaris\.org/bugdatabase/view\_bug\.do?bug\_id=6315357
> 
> 

Here's sort of an important question\.  Are you using the stock Solaris 10\, or are you using the 
modified and alpha OpenSolaris software?

**Comment by p5pRT** [comment]
### From @doughera88

On Tue\, 27 Jun 2006\, Marc Esser wrote&#8203;:

> don't mean to clobber the list\, but I just found a workaround for this 
> problem\. This got me past the message in configure \- so I should be good 
> to go for now\. Maybe include that in the README file as a known issue??
> 
> http&#8203;://bugs\.opensolaris\.org/bugdatabase/view\_bug\.do?bug\_id=6315357

Well\, that was marked as "Closed\, not a bug" back on 24\-Aug\-2005\, so I 
think it's safe to assume that it's not going to change\. Meanwhile\, I 
think I may have found the problem in the hints/solaris\_2\.sh file\.

On Solaris 10\, getconf returns \-xarch=generic64 on SPARC\.  That's a
valid cc flag\, as I pointed out before\, but not a valid assembler flag\.
What puzzled me was how the assembler was getting that flag\.  I thought
it was a gcc configuration issue\, but now I think it's a hint file
problem\.  I had even flagged it nearly 3 years ago and asked for testers
to verify\, but never heard anything back\.

The hints file adds \-m64 for use64bits\.  Then\, it also adds in
\-Wa\,xarch=generic64\, which gets passed to the assembler\.  I think that
gcc ought to handle the assembler flags itself\, but I have no way to
test it\.

Could you please try this patch and report back whether it works?


<details><summary>Inline Patch</summary>

```diff
diff -ru perl-current/hints/solaris_2.sh perl-andy/hints/solaris_2.sh
--- perl-current/hints/solaris_2.sh	2006-06-28 13:30:08.000000000 -0400
+++ perl-andy/hints/solaris_2.sh	2006-06-28 13:30:30.000000000 -0400
@@ -560,7 +560,7 @@
 		    # This adds in -Wa,-xarch=v9.  I suspect that's superfluous,
 		    # since the -m64 above should do that already.  Someone
 		    # with gcc-3.x.x, please test with gcc -v.   A.D. 20-Nov-2003
-		    ccflags="$ccflags -Wa,`getconf XBS5_LP64_OFF64_CFLAGS 2>/dev/null`"
+		    # ccflags="$ccflags -Wa,`getconf XBS5_LP64_OFF64_CFLAGS 2>/dev/null`"
 		fi
 		ldflags="$ldflags -m64"
 		lddlflags="$lddlflags -G -m64"

-- 
```
</details>

&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From @tux

On Wed\, 28 Jun 2006 13&#8203;:37&#8203;:54 \-0400 \(EDT\)\, Andy Dougherty
\<doughera@&#8203;lafayette\.edu> wrote&#8203;:

> On Tue\, 27 Jun 2006\, Marc Esser wrote&#8203;:
> 
> > don't mean to clobber the list\, but I just found a workaround for this 
> > problem\. This got me past the message in configure \- so I should be good 
> > to go for now\. Maybe include that in the README file as a known issue??
> > 
> > http&#8203;://bugs\.opensolaris\.org/bugdatabase/view\_bug\.do?bug\_id=6315357
> 
> Well\, that was marked as "Closed\, not a bug" back on 24\-Aug\-2005\, so I 
> think it's safe to assume that it's not going to change\. Meanwhile\, I 
> think I may have found the problem in the hints/solaris\_2\.sh file\.

Thanks\, applied as change \#28446

> On Solaris 10\, getconf returns \-xarch=generic64 on SPARC\.  That's a
> valid cc flag\, as I pointed out before\, but not a valid assembler flag\.
> What puzzled me was how the assembler was getting that flag\.  I thought
> it was a gcc configuration issue\, but now I think it's a hint file
> problem\.  I had even flagged it nearly 3 years ago and asked for testers
> to verify\, but never heard anything back\.
> 
> The hints file adds \-m64 for use64bits\.  Then\, it also adds in
> \-Wa\,xarch=generic64\, which gets passed to the assembler\.  I think that
> gcc ought to handle the assembler flags itself\, but I have no way to
> test it\.
> 
> Could you please try this patch and report back whether it works?
> 
> 
> diff \-ru perl\-current/hints/solaris\_2\.sh perl\-andy/hints/solaris\_2\.sh
> \-\-\- perl\-current/hints/solaris\_2\.sh    2006\-06\-28 13&#8203;:30&#8203;:08\.000000000 \-0400
> \+\+\+ perl\-andy/hints/solaris\_2\.sh    2006\-06\-28 13&#8203;:30&#8203;:30\.000000000 \-0400
> @&#8203;@&#8203; \-560\,7 \+560\,7 @&#8203;@&#8203;
>              \# This adds in \-Wa\,\-xarch=v9\.  I suspect that's superfluous\,
>              \# since the \-m64 above should do that already\.  Someone
>              \# with gcc\-3\.x\.x\, please test with gcc \-v\.   A\.D\. 20\-Nov\-2003
> \-            ccflags="$ccflags \-Wa\,\`getconf XBS5\_LP64\_OFF64\_CFLAGS 2>/dev/null\`"
> \+            \# ccflags="$ccflags \-Wa\,\`getconf XBS5\_LP64\_OFF64\_CFLAGS 2>/dev/null\`"
>          fi
>          ldflags="$ldflags \-m64"
>          lddlflags="$lddlflags \-G \-m64"
> 


\-\- 
H\.Merijn Brand        Amsterdam Perl Mongers \(http&#8203;://amsterdam\.pm\.org/\)
using & porting perl 5\.6\.2\, 5\.8\.x\, 5\.9\.x  on HP\-UX 10\.20\, 11\.00\, 11\.11\,
& 11\.23\, SuSE 10\.0\, AIX 4\.3 & 5\.2\, and Cygwin\.       http&#8203;://qa\.perl\.org
http&#8203;://mirrors\.develooper\.com/hpux/           http&#8203;://www\.test\-smoke\.org
&nbsp;                       http&#8203;://www\.goldmark\.org/jeff/stupid\-disclaimers/

**Comment by p5pRT** [comment]
### From esserm@us.ibm.com

stock

Marc Esser
IBM Global Services




"Steve Peters via RT" \<perlbug\-followup@&#8203;perl\.org> 
06/28/2006 09&#8203;:53 AM
Please respond to
perlbug\-followup@&#8203;perl\.org


To
Marc Esser/Encinitas/IBM@&#8203;IBMUS
cc

Subject
\[perl \#39636\] Configure fails






> \[esserm@&#8203;us\.ibm\.com \- Wed Jun 28 03&#8203;:54&#8203;:19 2006\]&#8203;:
> 
> Andy\,
> 
> don't mean to clobber the list\, but I just found a workaround for this 
> problem\. This got me past the message in configure \- so I should be good 

> to go for now\. Maybe include that in the README file as a known issue??
> 
> http&#8203;://bugs\.opensolaris\.org/bugdatabase/view\_bug\.do?bug\_id=6315357
> 
> 

Here's sort of an important question\.  Are you using the stock Solaris 10\, 
or are you using the 
modified and alpha OpenSolaris software?

**Comment by p5pRT** [comment]
p5p@spam.wizbit.be - Status changed from 'open' to 'resolved'
